Library/Homebrew/cmd/configure-shell.rb

```diff
@@ -0,0 +1,47 @@
+module Homebrew
+  module_function
+
+  def configure_shell
+    new_env_hash = create_new_env
+    new_env_file_contents = "# Homebrew environment\n# Created by running brew configure-shell\n"
+    # Write things from the new_env to new_env_file_contents
+    new_env_hash.each do |key, val|
+      next unless val
+      val = "1" if val == true
+      val = val.gsub '\\', '\\\\'
+      val = val.gsub '"', '\\"'
+
+      new_env_file_contents += "export #{key}=\"#{val}\"\n"
```

Comment:
You're coding this in a Bash-specific way... Look at the `export_value` function in `Library/Homebrew/utils/shell.rb` for an inspiration how to code this in a general way for any shell.
Another question: is the goal to make SHELL (Bash, Zsh, whatever) read this file, or Homebrew/brew?

Reply:
I mean, I understand that currently you're prepping it to make SHELL read this, but... should it be this way?

Reply:
Just as `/etc/environment` is always in `bash` syntax, I'm thinking that `$HOMEBREW_PREFIX/etc/environment` should similarly always be in `bash` syntax. If the user is using `csh` or `fish`, I feel the configuration should be written to `$HOMEBREW_PREFIX/etc/environment.csh` or `$HOMEBREW_PREFIX/etc/environment.fish`. This is necessary if the users uses more than one shell, say `bash` and `fish`. Note `zsh` is compatible with `bash` syntax.